services/QuillLMS/app/services/classlink_integration/instructor_linker.rb

```diff
@@ -11,17 +11,25 @@ def initialize(persisted_user:, classlink_instance_id:, school_sourced_id:)
     end
 
     def run
-      return unless persisted_user.role == ::User::ADMIN
+      return unless [::User::ADMIN, ::User::TEACHER].include?(persisted_user.role)
       return unless school_sourced_id
+      return unless classlink_instance_school
 
-      cis = ClasslinkInstanceSchool.find_by(classlink_instance_id:, external_id: school_sourced_id)
-      return unless cis
+      persisted_user.role == ::User::ADMIN ? create_admin_linkage : create_teacher_linkage
```

Comment:
I think this is a law of demeter violation.  Instead of reaching into the class for constants, use an existing method:

`persisted_user.admin?  ? create_admin_linkage : create_teacher_linkage`